code/datums/traits/negative.dm

```diff
@@ -496,19 +493,13 @@
 
 /datum/quirk/junkie/on_process()
 	var/mob/living/carbon/human/H = quirk_holder
-	if (tick_counter == 60) //Halfassed optimization, increase this if there's slowdown due to this quirk
-		var/in_list = FALSE
-		for (var/datum/reagent/entry in H.reagents.addiction_list)
-			if(istype(entry, reagent_type))
-				in_list = TRUE
-				break
-		if(!in_list)
+	if(world.time > next_process)
+		next_process = world.time + process_interval
+		if(!H?.reagents?.addiction_list?.Find(reagent_instance))
```

Comment:
These ?. are completely redundant if you access H.reagents.addiction_list right after.

Reply:
Yeah, good point, it was just some overzealous safety checking that didn't actually have to do with the runtime.

Reply:
the runtime is on the lines below, sometime reagent_instance gets nulled out (cloning?) and I'm not sure why exactly. This just creates a new instance if it's nulled.